docs/DeveloperGuide.md

```diff
@@ -2,12 +2,153 @@
+  application
++ Users may have differing frequencies for recurring events thus application gives them a few common options
+
+Illustrated below is the class diagram for the Recurrence Component:</br>
+</br>
+<img src="./Diagrams/recurrenceClassDiagram.png" alt="recurrenceClassDiagram" width="800"/>
+</br>
+</br>
+Illustrated below is the sequence diagram of the Recurrence Component: </br>
+</br>
+<img src="./Diagrams/recurrenceSequenceDiagram.png" alt="recurrenceSequenceDiagram" width="700"/>
+</br>
+For the reference fragment of 'load from storage', refer to [Storage component](#storage). </br>
+For the reference fragment of 'add recurring entry', refer to 
+[checkIncomeRecurrence / checkSpendingRecurrence](#checkincomerecurrence--checkspendingrecurrence-method) method. </br>
+
+
+#### How the Recurrence Component works:</br>
++ Upon running the application by the user, `Storage` component will load the `IncomeList` and `SpendingList` members of
+`Wiagi` to retrieve past data.
++ Both list are then iterated through. Each member of the list is parsed through `Parser` which returns the type of 
+recurrence it is (eg. `DailyRecurrence`, `null`) which is encapsulated as a `Recurrence` object.
++ If `Recurrence` is not `null` (ie. a recurring entry), it checks the entry and adds to the `SpendingList` and 
+`IncomeList` if needed. </br>
+
+#### Implementation:
+#### Recurrence class
+The `Recurrence` class is a abstract class that provides the interface for checking `Income` and `Spending` and adding 
+recurring entries into the list. </br>
+The following are the abstract methods defined: </br>
++ `checkSpendingRecurrence`
++ `checkIncomeRecurrence`
+
+The following are child classes of `Recurrence`:
++ `DailyRecurrence`: Handles entries labelled as daily recurring events
++ `MonthlyRecurrence`: Handles entries labelled as monthly recurring events
++ `YearlyRecurrence`: Handles entries labelled as yearly recurring events
+
+##### parseRecurrence method
+Class: `Parser` </br>
+Method Signature: </br>
+```
+public static Recurrence parseRecurrence(Type entry)
+```
+Functionality: </br>
+1. Takes in child class of `Type` (ie. `Spending`, `Income`)
+2. Matches the `reccurenceFrequency` attribute with switch case to determine which `Recurrence` child to return
+3. Returns `DaillyRecurrence`, `MonthlyRecurrence`, `YearlyRecurrence` or `null`(If not a recurring entry).
+
+##### checkIncomeRecurrence / checkSpendingRecurrence method
+Class: `DailyRecurrence`, `MonthlyRecurrence`, `YearlyRecurrence` </br>
+Method Signature: </br>
+```
+@Override
+public void checkIncomeRecurrence(Income recurringIncome, IncomeList incomes)
+@Override
+public void checkSpendingRecurrence(Spending recurringSpending, SpendingList spendings)
+```
+Below illustrates the functionality of the checkIncomeRecurrence method through a sequence diagram </br>
+</br>
+<img src="./Diagrams/addRecurrenceEntry.png" alt="addRecurrenceEntry" width="700"/> </br>
```

Comment:
Instead of using the actual operators, potentially use what it intuitively means instead. (e.g. next recurrence is due). 
This could help to reduce the complexity of the diagram.
<img width="368" alt="image" src="https://github.com/user-attachments/assets/13a544a2-1328-4bdc-bfcc-0867f966956f">

Reply:
I like that you used a note to explain CheckDate. 
Maybe consider choosing a more descriptive variable name for CheckDate instead. Could help to improve the readability of the code and your diagrams because fewer additional comments or notes are relevant.